crates/cli/src/commands/build.rs

```diff
@@ -21,225 +22,432 @@ use openvm_transpiler::{elf::Elf, openvm_platform::memory::MEM_SIZE};
+    println!("[openvm] Manifest directory: {}", manifest_dir.display());
+
+    // Get target path
+    let target_path = if let Some(target_path) = &cargo_args.target_dir {
+        target_path.to_path_buf()
+    } else {
+        get_target_dir(&manifest_path)
+    };
+
+    // Set guest options using build arguments; use found manifest directory for consistency
     let mut guest_options = GuestOptions::default()
-        .with_features(build_args.features.clone())
-        .with_profile(build_args.profile.clone())
+        .with_features(cargo_args.features.clone())
+        .with_profile(cargo_args.profile.clone())
         .with_rustc_flags(var("RUSTFLAGS").unwrap_or_default().split_whitespace());
-    guest_options.target_dir = build_args.target_dir.clone();
-    if build_args.offline {
-        guest_options.options = vec!["--offline".to_string()];
+
+    guest_options.target_dir = Some(target_path);
+    guest_options
+        .options
+        .push(format!("--color={}", cargo_args.color));
+    guest_options.options.push("--manifest-path".to_string());
+    guest_options
+        .options
+        .push(manifest_path.to_string_lossy().to_string());
+
+    for pkg in &cargo_args.package {
+        guest_options.options.push("--package".to_string());
+        guest_options.options.push(pkg.clone());
+    }
+    for pkg in &cargo_args.exclude {
+        guest_options.options.push("--exclude".to_string());
+        guest_options.options.push(pkg.clone());
+    }
+    for target in &cargo_args.bin {
+        guest_options.options.push("--bin".to_string());
+        guest_options.options.push(target.clone());
+    }
+    for example in &cargo_args.example {
+        guest_options.options.push("--example".to_string());
+        guest_options.options.push(example.clone());
+    }
+
+    let all_bins = cargo_args.bins || cargo_args.all_targets;
+    let all_examples = cargo_args.examples || cargo_args.all_targets;
+
+    let boolean_flags = [
+        ("--workspace", cargo_args.workspace),
+        ("--lib", cargo_args.lib),
+        ("--bins", all_bins),
+        ("--examples", all_examples),
+        ("--all-features", cargo_args.all_features),
+        ("--no-default-features", cargo_args.no_default_features),
+        ("--verbose", cargo_args.verbose),
+        ("--quiet", cargo_args.quiet),
+        ("--ignore-rust-version", cargo_args.ignore_rust_version),
+        ("--locked", cargo_args.locked),
+        ("--offline", cargo_args.offline),
+        ("--frozen", cargo_args.frozen),
+    ];
+    for (flag, enabled) in boolean_flags {
+        if enabled {
+            guest_options.options.push(flag.to_string());
+        }
     }
 
+    // Build (allowing passed options to decide what gets built)
+    let target_dir = build_generic(&guest_options)?;
+    if cargo_args.lib {
+        return Ok(None);
+    }
+
+    // Write to init file
     let app_config = read_config_toml_or_default(&build_args.config)?;
```

Comment:
noting that this means we will try to use the same openvm.toml for all build targets within this command. I thought about it and it does seem the most reasonable option: if you wanted to compile just a select target with a different config, you should run the command with the right target and `--config` custom